documentation/blog/2025-10-03-running-goose-in-containerized-environments/index.md

```diff
@@ -184,78 +183,141 @@ analyze:
 
 I've used this setup on multiple projects, and it's a game-changer for automated code reviews and documentation generation.
 
-## Troubleshooting: Common Gotchas
+## Troubleshooting: Real-World Issues
 
-### Permission Issues
+### File Permission Problems with Workspace Mounts
 
-If you see permission errors with mounted volumes, match the user ID:
+When goose can't write to your mounted workspace, it's usually a user ID mismatch:
 
 ```bash
-docker run --rm \
-    -v $(pwd):/workspace \
-    -u $(id -u):$(id -g) \
-    goose:local run -t "List files"
+# Problem: "Permission denied" when goose tries to create files
+$ docker run --rm -v $(pwd):/workspace ghcr.io/block/goose:v0.9.3
+
+# Solution: Match container user with host user
+$ docker run --rm \
+   -v $(pwd):/workspace \
+   -u $(id -u):$(id -g) \
+   ghcr.io/block/goose:v0.9.3 run -t "Create a README.md"
 ```
 
-### API Key Headaches
+### Managing Multiple API Keys
 
-Make sure your environment variables are set correctly. Use `--env-file` for multiple keys:
+When working with multiple LLM providers (OpenAI, Anthropic, Google), passing individual `-e` flags becomes unwieldy and error-prone. Instead, use a single `.env` file:
 
 ```bash
-echo "OPENAI_API_KEY=your-key-here" > .env
-docker run --rm --env-file .env goose:local run -t "Test command"
+# Create a comprehensive .env file
+$ cat > .env << EOF
+GOOSE_PROVIDER=openai
+GOOSE_MODEL=gpt-4o
+OPENAI_API_KEY=sk-your-openai-key
+ANTHROPIC_API_KEY=sk-ant-your-anthropic-key
+GOOGLE_API_KEY=your-google-key
+EOF
+
+# Use the entire file at once
+$ docker run --rm --env-file .env -v $(pwd):/workspace goose:v0.9.3
 ```
 
-### Network Access
+This approach is cleaner, more maintainable, and essential for CI/CD environments where you might switch between providers based on cost, availability, or model capabilities. It also keeps sensitive keys out of your shell history and makes it easy to version different configurations for different environments.
+
+### Connecting to Local Development Services
+
+When goose needs to interact with local databases, development servers, or APIs running on your host machine, the container's isolated network becomes a barrier. This is common when:
 
-For local services, use host networking:
+- Testing integrations with a local PostgreSQL database
```

Comment:
this bullet points are mostly covered in the intro sentence above, and can be removed.